azure-pipelines.yml

```diff
@@ -1,11 +1,96 @@
-# Starter pipeline
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
 # https://aka.ms/yaml
+variables:
+  # TODO: Set this where necessary
+  travis_event_type: ''
 
-pool:
-  vmImage: 'Ubuntu 16.04'
+jobs:
```

Comment:
Do these jobs just mirror the Travis ones? If so, do they work the same way - i.e. failure in early ones blocks later ones?

Reply:
These don't entirely mirror the Travis ones-- my intent was to have all these checks run in parallel and then use the [dependsOn stanza](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/multiple-phases?view=azure-devops&tabs=yaml) to block deploys later. In general Azure Pipelines doesn't do any implicit ordering or dependencies and here I'm defaulting to 'all tests errors may be useful'. 

Should we introduce ordering/dependencies for these jobs?

Reply:
Hmm. Depends how much parallelism we get between runs? The main motivation for the split in Travis is that if the fast checks fail you're almost certainly going to want to push a new version soon and there's no point hogging the build queue waiting for the slow ones to finish. If different runs are still competing for resources (which I'd assume they are) and it's easy to implement something similar, it'd be great if we could.

Reply:
We have a total of ten agents at any time, so there can definitely still be resource contention and it's really straightforward to match Travis' functionality. There might be a couple of performance optimizations available here from the increased control:

1) We can definitely have whole-repo-tests run concurrently in its own job instead of blocking downstream for 15-20 minutes
2) We /might/ be able to have jobs startup IFF upstream dependencies have either succeeded or are running and there's room

I think I'd prefer the second option since it helps address the resource contention angle while also allowing for the tests to take advantage of all the parallelism available to them, but I'll have to see if that's possible first.